packages/postman-to-openapi/src/convert.ts

```diff
@@ -0,0 +1,771 @@
+import type { OpenAPIV3 } from '@scalar/openapi-types'
+import { parse } from 'jsonc-parser'
+import camelCase from 'lodash.camelcase'
+
+import type {
+  Options,
+  PostmanUrl,
+  PostmanVariable,
+  SecurityDefinition,
+} from './types'
+
+function parseMdTable(): Record<string, any> {
+  return {}
+}
+
+function replacePostmanVariables(collectionFile: any): any {
+  return collectionFile
+}
+/**
+ * Converts a Postman collection to an OpenAPI specification.
+ *
+ * @param {string | Record<string, any>} collectionContent - The Postman collection content, either as a JSON string or an object.
+ * @param {Options} [options={}] - Configuration options for the conversion process.
+ * @returns {Promise<string>} A promise that resolves to the OpenAPI specification as a JSON string.
+ */
+export async function convert(
+  collectionContent: string | Record<string, any>,
+  options: Options = {},
+): Promise<string> {
+  const {
+    info = {},
+    defaultTag = 'default',
+    pathDepth = 0,
+    auth: optsAuth,
+    servers,
+    externalDocs = {},
+    folders = {},
+    responseHeaders = true,
+    replaceVars = false,
+    disabledParams = { includeQuery: false, includeHeader: false },
+    operationId = 'off',
+  } = options
+
+  let collectionFile: Record<string, any>
+  if (typeof collectionContent === 'string') {
+    collectionFile = JSON.parse(collectionContent)
+  } else {
+    collectionFile = collectionContent
+  }
+  if (replaceVars) {
+    collectionFile = replacePostmanVariables(collectionFile)
+  }
+  const postmanJson = collectionFile.collection || collectionFile
+  const { item: items = [], variable = [] } = postmanJson
+  const paths: Record<string, any> = {}
+  const domains = new Set<string>()
+  const tags: Record<string, any> = {}
+  const securitySchemes: Record<string, any> = {}
+
+  for (let i = 0; i < items.length; i++) {
+    let element = items[i]
+    while (element?.item != null) {
+      const { item = [], description: tagDesc = '' } = element
+      const tag = calculateFolderTag(element, folders)
+      const tagged = item.map((e: any) => ({ ...e, tag }))
+      tags[tag] = tagDesc
+      items.splice(i, 1, ...tagged)
+      element = tagged.length > 0 ? tagged.shift() : items[i]
+    }
```

Comment:
While loops can be difficult to follow and can have undefined exit conditions. 
We are also mutating the array we are iterating over here (items) which will cause unintended side effects (and is a bad code smell).